Model/Api/CreateOrder.php

```diff
@@ -451,42 +462,44 @@ public function validateCartItems($quote, $transaction)
             $productId = $item->getProductId();
             $itemPrice = $this->cartHelper->getRoundAmount($item->getPrice());
 
-            $this->validateItemInventory($sku, $transactionItemsSkuQty[$sku], $productId);
+            $this->hasItemErrors($item);
             $this->validateItemPrice($sku, $itemPrice, $transactionItems);
         }
     }
 
     /**
-     * @param string $itemSku
-     * @param int $itemQty
-     * @param int $productId
+     * Check if quote have errors.
+     *
+     * @param $quoteItem
+     * @return bool
      * @throws BoltException
-     * @throws NoSuchEntityException
      */
-    public function validateItemInventory($itemSku, $itemQty, $productId)
+    public function hasItemErrors($quoteItem)
     {
-        $stockItem = $this->stockRegistry->getStockItem($productId);
+        if (empty($quoteItem->getErrorInfos())) {
+            return true;
+        }
 
-        if (!$stockItem->getManageStock()) return;
+        $errorInfos = $quoteItem->getErrorInfos();
+        $message = '';
+        foreach ($errorInfos as $errorInfo) {
+            $message .= '(' . $errorInfo['origin'] . '): ' . $errorInfo['message']->render() . PHP_EOL;
+        }
 
-        $inStock = $stockItem->getIsInStock();
-        $stockQty = $stockItem->getQty();
+        $this->bugsnag->registerCallback(function ($report) use ($message) {
+            $report->setMetaData([
+                'Pre Auth' => [
+                    'quoteItem errors' => $message,
+                ]
+            ]);
+        });
+        throw new BoltException(
+            __($message),
+            null,
+            self::E_BOLT_GENERAL_ERROR
```

Comment:
Can we throw E_BOLT_ITEM_OUT_OF_INVENTORY error here? Vast majority if not all errors we catch should be exactly that. Or explicitly check $errorInfo code property to match ERROR_QTY, vendor/magento/module-catalog-inventory/Helper/Data.php

Reply:
Updated